src/RosClientNode.hpp

```diff
@@ -38,24 +51,42 @@ class RosClientNode : public QObject {
    * @param msg_type type of the message; can be obtained using
    * ros::message_traits::DataType
    * @param from_topic topic on which message was received
+   * @param to_topic remote topic to send to
    */
   template <typename T>
   void encode_ros_msg(
-      const T& msg, const std::string& msg_type, const std::string& to_topic) {
-    auto& rate_info = rate_limits[to_topic];
-    const auto curr_time = std::chrono::high_resolution_clock::now();
-    const auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(curr_time - rate_info.first);
-    const double elapsed_sec = duration.count() / 1000.0;
-
-    if (elapsed_sec > rate_info.second) {
-      rate_info.first = curr_time;
-      flatbuffers::FlatBufferBuilder fbb;
-      auto metadata = encode_metadata(fbb, msg_type, to_topic);
-      auto root_offset = encode<T>(fbb, msg, metadata);
-      fbb.Finish(flatbuffers::Offset<void>(root_offset));
-      Q_EMIT ros_message_encoded(QByteArray(
-          reinterpret_cast<const char*>(fbb.GetBufferPointer()), fbb.GetSize()));
+      const T& msg, const std::string& msg_type, const std::string& from_topic,
+      const std::string& to_topic) {
+    // check rate limits
+    if (rate_limits.count(from_topic)) {
```

Comment:
Hmm, how do we feel about having both rate_limits and this priority queue thing?
They both were with the goal of mitigating overuse of the network, and while I can see use cases for both (specfiically very heavyweight messages), I wonder if keeping both around will be confusing in terms of config etc.

Reply:
I think there could still be use cases for them, so I'm going to leave the functionality there. I'll make sure to clarify in the docs that rate limits are optional.